CircuitPython_Pyloton/pyloton.py

```diff
@@ -0,0 +1,563 @@
+    WHITE = 0xFFFFFF
+
+    start = time.time()
+
+    track_artist = True
+
+    clue = Clue()
+
+    def __init__(self, ble, display, circ, heart=True, speed=True, cad=True, ams=True, debug=False):
+        self.debug = debug
+
+        self.ble = ble
+
+        self.display = display
+
+        self.circumference = circ
+
+        self.heart_enabled = heart
+        self.speed_enabled = speed
+        self.cadence_enabled = cad
+        self.ams_enabled = ams
+
+        self.hr_connection = None
+
+        self.num_enabled = heart + speed + cad + ams
+        self._load_fonts()
+
+        self.sprite_sheet, self.palette = adafruit_imageload.load("/sprite_sheet.bmp",
+                                                                  bitmap=displayio.Bitmap,
+                                                                  palette=displayio.Palette)
+
+
+    def show_splash(self):
+        """
+        Shows the loading screen
+        """
+        if self.debug:
+            return
+        with open('blinka-pyloton.bmp', 'rb') as bitmap_file:
+            bitmap1 = displayio.OnDiskBitmap(bitmap_file)
+
+            tile_grid = displayio.TileGrid(bitmap1, pixel_shader=displayio.ColorConverter())
+
+
+            self.loading_group.append(tile_grid)
+
+            self.display.show(self.loading_group)
+
+            status_heading = label.Label(font=self.arial16, x=80, y=175,
+                                         text="Status", color=self.YELLOW)
+
+            rect = Rect(0, 165, 240, 75, fill=self.PURPLE)
+
+            self.loading_group.append(rect)
+            self.loading_group.append(status_heading)
+
+            self.display.show(self.loading_group)
+            time.sleep(.01)
+
+
+    def _load_fonts(self):
+        """
+        Loads fonts
+        """
+        self.arial12 = bitmap_font.load_font("/fonts/Arial-12.bdf")
+        self.arial16 = bitmap_font.load_font("/fonts/Arial-16.bdf")
+        self.arial24 = bitmap_font.load_font("/fonts/Arial-Bold-24.bdf")
+
+
+    def _status_update(self, message):
+        """
+        Displays status updates
+        """
+        if self.debug:
+            print(message)
+            return
+
+        text_group = displayio.Group()
+        if len(message) > 25:
+            status = label.Label(font=self.arial12, x=10, y=200,
+                                 text=message[:25], color=self.YELLOW)
+            status1 = label.Label(font=self.arial12, x=10, y=220,
+                                  text=message[25:], color=self.YELLOW)
+
+            text_group.append(status)
+            text_group.append(status1)
+        else:
+            status = label.Label(font=self.arial12, x=10, y=200, text=message, color=self.YELLOW)
+            text_group.append(status)
+
+
+
+        if len(self.loading_group) < 4:
+            self.loading_group.append(text_group)
+        else:
+            self.loading_group[3] = text_group
+
+        self.display.show(self.loading_group)
```

Comment:
Could you create the labels and groups just once and reuse them?

Reply:
@dhalbert The reason I was doing it this way was that sometimes it would be one line and sometimes it would be two. But, I do agree that doing it your way would be cleaner and probably clean things up a bit.